drivers/include/can/candev.h

```diff
@@ -0,0 +1,203 @@
+};
+
+/**
+ * @brief Structure to hold driver interface -> function mapping
+ */
+typedef struct candev_driver {
+    /**
+     * @brief Send packet
+     *
+     * @param[in] dev       CAN device descriptor
+     * @param[in] frame     CAN frame to send
+     *
+     * @return < 0 on error
+     * @return mailbox id >= 0 if OK
+     */
+    int (*send)(candev_t *dev, const struct can_frame *frame);
+
+    /**
+     * @brief Abort a packet sending
+     *
+     * @param[in] dev       CAN device descriptor
+     * @param[in] frame     CAN frame to abort
+     *
+     * @return < 0 on error
+     * @return 0 on OK
+     */
+    int (*abort)(candev_t *dev, const struct can_frame *frame);
+
+    /**
+     * @brief the driver's initialization function
+     *
+     * @param[in] dev       CAN device descriptor
+     *
+     * @return < 0 on error, 0 on success
+     */
+    int (*init)(candev_t *dev);
+
+    /**
+     * @brief a driver's user-space ISR handler
+     *
+     * @param[in] dev       CAN device descriptor
+     */
+    void (*isr)(candev_t *dev);
+
+    /**
+     * @brief   Get an option value from a given CAN device
+     *
+     * @param[in]   dev     CAN device descriptor
+     * @param[in]   opt     option type
+     * @param[out]  value   pointer to store the option's value in
+     * @param[in]   max_len maximal amount of byte that fit into @p value
+     *
+     * @return              number of bytes written to @p value
+     * @return              <0 on error
+     */
+    int (*get)(candev_t *dev, canopt_t opt, void *value, size_t max_len);
+
+    /**
+     * @brief   Set an option value for a given CAN device
+     *
+     * @param[in] dev       CAN device descriptor
+     * @param[in] opt       option type
+     * @param[in] value     value to set
+     * @param[in] value_len the length of @p value
+     *
+     * @return              number of bytes used from @p value
+     * @return              <0 on error
+     */
+    int (*set)(candev_t *dev, canopt_t opt, void *value, size_t value_len);
+
+    /**
+     * @brief   Set a receive @p filter
+     *
+     * @param[in] dev       CAN device descriptor
+     * @param[in] filter    filter to set
+     *
+     * @return              a positive filter number
+     * @return              <0 on error
+     */
+    int (*set_filter)(candev_t *dev, const struct can_filter *filter);
+
+    /**
+     * @brief  Remove a @p filter
+     *
+     * @param[in] dev       CAN device descriptor
+     * @param[in] filter    filter to remove
+     *
+     * @return              0 on success
+     * @return              <0 on error
+     */
+    int (*remove_filter)(candev_t *dev, const struct can_filter *filter);
+
+    /**
+     * @brief  Power up the given CAN device
+     *
+     * @param[in] dev       CAN device descriptor
+     *
+     * @return              0 on success
+     * @return              <0 on error
+     */
+    int (*power_up)(candev_t *dev);
```

Comment:
but is it my diff view or do the `power_xx` functions still exist in this interface?!